pyiron/vasp/structure.py

```diff
@@ -177,44 +178,51 @@ def atoms_from_string(string, read_velocities=False, species_list=None):
     n_atoms = 0
     for key in atoms_dict["species_dict"].keys():
         n_atoms += atoms_dict["species_dict"][key]["count"]
-    for i in range(position_index, position_index + n_atoms):
-        vec = list()
-        for j in range(3):
-            vec.append(float(string[i].split()[j]))
-        if atoms_dict["selective_dynamics"]:
-            selective_dynamics.append(["T" in string[i].split()[k] for k in range(3, 6)])
-        positions.append(vec)
+    try:
+        for i in range(position_index, position_index + n_atoms):
+            vec = list()
+            for j in range(3):
+                vec.append(float(string[i].split()[j]))
+            if atoms_dict["selective_dynamics"]:
+                selective_dynamics.append(["T" in string[i].split()[k] for k in range(3, 6)])
+            positions.append(vec)
+    except (ValueError, IndexError):
```

Comment:
Do we need an exception here or can we check the shape before?

Reply:
The issue is for incomplete/corrupted files where we can't really check the shape unless we parse it. And it throws an `IndexError` or `ValueError` when parsing

Reply:
Can not you check the length of the string? in addition you seem to execute the split multiple times during the for loop, it would be more efficient to split it once and store it in a variable.

Reply:
Can not we do it like this? https://github.com/pyiron/pyiron/pull/133/commits/a6be412fb3d582891e59fe0763792c3b06b67511

Reply:
I think 2 `if` statements inside a for loop slows things down a lot when parsing large files. Wouldn't a `try` and `except` be much faster?

Reply:
From my perspective raising an exception is a lot worth in terms of code style.